service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceOperations.java

```diff
@@ -214,9 +197,95 @@ public static HttpResponse request(
         request.setHeaderField(ACCEPT, ACCEPT_VALUE);
         request.setHeaderField(CONTENT_TYPE, ACCEPT_VALUE + "; " + ACCEPT_CHARSET);
 
+        if (payload != null)
+        {
+            request.setHeaderField(CONTENT_LENGTH, String.valueOf(payload.length));
```

Comment:
👍 
Is this optional and that's why did not exist before?

Reply:
It is optional, but I'm surprised we don't send it anyways. It is very typical to include this header in every HTTP request